aesara/link/basic.py

```diff
@@ -641,7 +641,8 @@ def create_jitable_thunk(
             The JITed function that performs the computations.
 
         """
-        output_nodes = [o.owner for o in self.fgraph.outputs]
+        # This is a bit hackish, but we only return one of the output nodes
+        output_nodes = [o.owner for o in self.fgraph.outputs if o.owner is not None][:1]
```

Comment:
This wasn't changed by this PR, but I wonder does the hack actually work? There's a whole bunch of logic downstream concerning garbage collection and whatnot that is supposed to rely on this. Too dense for me to take a look though :D

Reply:
> This wasn't changed by this PR, but I wonder does the hack actually work? There's a whole bunch of logic downstream concerning garbage collection and whatnot that is supposed to rely on this. Too dense for me to take a look though :D

It must be working; how else have we been using Numba/JAX to evaluate graphs?  

The hackish part is that we were and still are (to some extent) using the old `Function`/`VM` framework to do something that it wasn't designed to do.  For example, **we don't have a thunk for each node in the graph**, like we normally would.

To be clear, shared variable and `updates` support for the JIT backends have always been incomplete, and this falls squarely into the latter.